x/superfluid/keeper/stake_test.go

```diff
@@ -408,6 +408,152 @@ func (suite *KeeperTestSuite) TestSuperfluidUnbondLock() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestSuperfluidUndelegateAndUnbondLock() {
+	var lockAmount int64 = 1000000
+	testCases := []struct {
+		name            string
+		testInvalidLock bool
+		unlockAmount    sdk.Int
+		expectErr       bool
+		splitLockId     bool
+	}{
+		{
+			name:            "lock doesn't exist",
+			testInvalidLock: true,
+			unlockAmount:    sdk.NewInt(0),
+			expectErr:       true,
+			splitLockId:     false,
+		},
+		{
+			name:            "unlock amount = 0",
+			testInvalidLock: false,
+			unlockAmount:    sdk.NewInt(0),
+			expectErr:       true,
+			splitLockId:     false,
+		},
+		{
+			name:            "unlock amount > locked amount",
+			testInvalidLock: false,
+			unlockAmount:    sdk.NewInt(lockAmount + 1),
+			expectErr:       true,
+			splitLockId:     false,
+		},
+		{
+			name:            "lock is not split if unlock amount = locked amount",
+			testInvalidLock: false,
+			unlockAmount:    sdk.NewInt(lockAmount),
+			expectErr:       false,
+			splitLockId:     false,
+		},
+		{
+			name:            "lock is split if unlock amount < locked amount",
+			testInvalidLock: false,
+			unlockAmount:    sdk.NewInt(lockAmount / 2),
+			expectErr:       false,
+			splitLockId:     true,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			// setup validators
+			valAddrs := suite.SetupValidators([]stakingtypes.BondStatus{stakingtypes.Bonded})
+
+			denoms, _ := suite.SetupGammPoolsAndSuperfluidAssets([]sdk.Dec{sdk.NewDec(20), sdk.NewDec(20)})
+
+			// setup superfluid delegations
+			_, intermediaryAccs, locks := suite.setupSuperfluidDelegations(valAddrs, []superfluidDelegation{{0, 0, 0, lockAmount}}, denoms)
+			suite.checkIntermediaryAccountDelegations(intermediaryAccs)
+			suite.Require().True(len(locks) > 0)
```

Comment:
isn't lock length always = 1 ? If so, can we put it as a variable and then remove the for loop below/

Reply:
Yes I can. Alternatively would you like me to test with multiple locks?